### PR TITLE
feat: Shared encryption active keys table

### PR DESCRIPTION
**Pull request by yaelAbergel**

### CODE REVIEW

File: crates/sui-framework/packages/dwallet-system/sources/dwallet.move [hunk 1]
```diff
@@ -451,17 +453,50 @@ module dwallet_system::dwallet {
...
         transfer::freeze_object(encryption_key);
-        encryption_key_id
+    }
+
+    struct ActiveEncryptionKeys has key {
+        id: UID,
+        encryption_keys: Table<address, ID>,
+    }
+
+    public fun create_active_encryption_keys(ctx: &mut TxContext) {
```
Comment (ycscaly) [inline crates/sui-framework/packages/dwallet-system/sources/dwallet.move:472]: I think this function should be called once during initialization of this module

File: crates/sui-framework/packages/dwallet-system/sources/dwallet.move [hunk 2]
```diff
@@ -451,17 +453,50 @@ module dwallet_system::dwallet {
...
+        };
+        transfer::share_object(holder);
+    }
+
+    public fun set_active_encryption_key(
+        encryption_key_holder: &mut ActiveEncryptionKeys,
+        encryption_key: &EncryptionKey,
+        _ctx: &mut TxContext
+    ) {
+        if (table::contains(&encryption_key_holder.encryption_keys, encryption_key.key_owner_address)) {
```
Comment (ycscaly) [inline crates/sui-framework/packages/dwallet-system/sources/dwallet.move:485]: Check first that the caller is the key owner address

**Review by ycscaly — CHANGES_REQUESTED** [review]
Also, add please a check that when you call `encrypt_user_share` to encrypt to someone else (i.e. not the caller) it is the active encryption share

File: crates/sui-framework/packages/dwallet-system/sources/dwallet.move (hunk 1 above)
Comment (yaelAbergel) [reply crates/sui-framework/packages/dwallet-system/sources/dwallet.move:472]: This is correct and aligns with the initial idea, but Omer mentioned that it's too complex to implement in Move at this stage and should be addressed separately in the future.

**Comment by yaelAbergel** [comment]
> Also, add please a check that when you call `encrypt_user_share` to encrypt to someone else (i.e. not the caller) it is the active encryption share

Do we eventually want to limit the user? @omersadika @ycscaly

File: crates/sui-framework/packages/dwallet-system/sources/dwallet.move (hunk 1 above)
Comment (ycscaly) [reply crates/sui-framework/packages/dwallet-system/sources/dwallet.move:473]: Please open an issue for this

Comment (ItayLevyOfficial) [reply crates/sui-framework/packages/dwallet-system/sources/dwallet.move:473]: https://github.com/dwallet-labs/dwallet-network/issues/157

**Review by ycscaly — APPROVED** [review]

File: crates/sui-framework/packages/dwallet-system/sources/dwallet.move [hunk 3]
```diff
@@ -467,17 +470,52 @@ module dwallet_system::dwallet {
...
         transfer::freeze_object(encryption_key);
-        encryption_key_id
+    }
+
+    struct ActiveEncryptionKeys has key {
+        id: UID,
+        encryption_keys: Table<address, ID>,
+    }
+
+    // TODO: make private and call from within the init, find a way to get the object id - https://github.com/dwallet-labs/dwallet-network/issues/157
```
Comment (zeevmoney) [inline crates/sui-framework/packages/dwallet-system/sources/dwallet.move:489]:
```suggestion
```

File: crates/sui-framework/packages/dwallet-system/sources/dwallet.move [hunk 4]
```diff
@@ -467,17 +470,52 @@ module dwallet_system::dwallet {
...
+    public fun create_active_encryption_keys(ctx: &mut TxContext) {
```
Comment (zeevmoney) [inline crates/sui-framework/packages/dwallet-system/sources/dwallet.move:490]: Doc

File: crates/sui-framework/packages/dwallet-system/sources/dwallet.move [hunk 5]
```diff
@@ -467,17 +470,52 @@ module dwallet_system::dwallet {
...
+        let holder = ActiveEncryptionKeys {
+            id: object::new(ctx),
+            encryption_keys: table::new(ctx),
+        };
+        transfer::share_object(holder);
+    }
+
+    public fun set_active_encryption_key(
```
Comment (zeevmoney) [inline crates/sui-framework/packages/dwallet-system/sources/dwallet.move:498]: Doc

File: crates/sui-framework/packages/dwallet-system/sources/dwallet.move [hunk 6]
```diff
@@ -467,17 +470,52 @@ module dwallet_system::dwallet {
...
+            table::remove(&mut encryption_key_holder.encryption_keys, encryption_key.key_owner_address);
+        };
+        table::add(
+            &mut encryption_key_holder.encryption_keys,
+            encryption_key.key_owner_address,
+            object::id(encryption_key)
+        );
+    }
+
+    public fun get_active_encryption_key(
```
Comment (zeevmoney) [inline crates/sui-framework/packages/dwallet-system/sources/dwallet.move:514]: Doc

**Review by zeevmoney — CHANGES_REQUESTED** [review]

**Comment by zeevmoney** [comment]
@yaelAbergel fix conflicts.

**Pull request merged**